oap-server/server-query-plugin/query-graphql-plugin/src/main/java/org/apache/skywalking/oap/query/graphql/resolver/MetadataQueryV2.java

```diff
@@ -67,6 +67,10 @@ public List<Service> listServices(final String layer) throws IOException {
         return getMetadataQueryService().listServices(layer, null);
     }
 
+    public List<Service> listGroupServices(final String layer, final String group) throws IOException {
```

Comment:
Where is the approval of this new query API? I can't find this.

Reply:
> And what is the reason of ElasticSearch change? A pull request must include the explanation. One pull request is about one thing. Please be clear about the change, otherwise, we don't know what do you mean, and how could we follow?

Sorry this the first to pull request for open source ,I explain will explain it carefully.

Reply:
> > And what is the reason of ElasticSearch change? A pull request must include the explanation. One pull request is about one thing. Please be clear about the change, otherwise, we don't know what do you mean, and how could we follow?
> 
> Sorry this the first to pull request for open source ,I explain will explain it carefully.

You are replying on another context. Please the keep thread clear.

Reply:
> Where is the approval of this new query API? I can't find this.

I add the group and layer search condition,because the performance for oap server is very sufficient.In product,I as the user with 10000 agent and use the group classify the search result will be limit by the query max size with min(metadataSize and scrollsize) with 5000,but the scroll query will help the user to search as many as the user want service to to serarch by the agent. In order to let the user to find the service sufficient,I want to modify the code.
1.I give the user to find the service by group and layer when the user have classify the service by group.
2.I give the user to query the service as they want the number they want but not limit by the elasticsearch maxWindowSize.
So I modify the word by change the two place,I think carefully and without to change the other,I think this will very flexible and will more sufficient.

Reply:
In product,If the user change the metadataSize 、scrollsize over the elasticserch maxWindowSize.  ,It also will be limit by  the elasticserch maxWindowSize and give an exception. So I change the limit logical to use the scrollSize and metadataSize .This change will not cause the exception.

Reply:
> .In product,I as the user with 10000 agent and use the group classify the search result will be limit by the query max size with min(metadataSize and scrollsize) with 5000,

I can't follow this. 10k agents only mean 10k instances. Service must be much less than that.

Reply:
In fact the skywalking ui include  many different agent, kubenetes,istio and many, but the user may by just use the java agent service or the python agent, so they will write a ui by they use,the new query API wll helpfull for this.

Reply:
> metadataSize 、scrollsize over the elasticserch maxWindowSize

All these are open for you to change. Why don't you change them by your own?

Reply:
> In fact the skywalking ui include many different agent, kubenetes,istio and many, but the user may by just use the java agent service or the python agent, so they will write a ui by they use,the new query API wll helpfull for this.

So what? I can't see in any cases, you are limited by this. Set them as the number you want ,

Reply:
> > .In product,I as the user with 10000 agent and use the group classify the search result will be limit by the query max size with min(metadataSize and scrollsize) with 5000,
> 
> I can't follow this. 10k agents only mean 10k instances. Service must be much less than that.

In product, the metrics-all index will generate the data by day ,the service with the same name is added,does I setting something wrong for oap server.I  query the elasticsearch for index metrics-all and find the data is that.

Reply:
> > metadataSize 、scrollsize over the elasticserch maxWindowSize
> 
> All these are open for you to change. Why don't you change them by your own?

I change the metadataSize 20000 but this param is also use by the other query ,this cause an exception.This is over the elasticserch limit 10000. I also try to change the elasticsearch maxWindowSize to 30000,but this is very low performance for the query,So I think to split the metadataSize  and maxWindowSize to by this.

Reply:
> In product, the metrics-all index will generate the data by day ,the service with the same name is added

No, they are not. Service would be only added once. Check the data before guessing. That is not a fact.

Reply:
> Where is the approval of this new query API? I can't find this.



> > > And what is the reason of ElasticSearch change? A pull request must include the explanation. One pull request is about one thing. Please be clear about the change, otherwise, we don't know what do you mean, and how could we follow?
> > 
> > 
> > Sorry this the first to pull request for open source ,I explain will explain it carefully.
> 
> You are replying on another context. Please the keep thread clear.

1.This will help the query to service by group which will classify the service by group,which had many services.This will more sufficient and flexible.
2.Split the metadataSize and resultWindowMaxSize and aviod an exception when the user did not change elasticserch max size limit. And will find the query result as many as the user want to search,without reduce performance。

Reply:
And your explanation doesn't make the sense even for your codes. 
You added `resultWindowMaxSize`, but only `Math.min(metadataMaxSize, resultWindowMaxSize)`. Why just set a larger `metadataMaxSize`? `resultWindowMaxSize` isn't used in any place of this DAO. 
Still in all other places, `queryMaxSize`(`metadataMaxSize`) is still used. What is the point? What is changed?

Reply:
> 2.Split the metadataSize and resultWindowMaxSize and aviod an exception when the user did not change elasticserch max size limit. And will find the query result as many as the user want to search,without reduce performance。

I don't know how you get this conclusion. There is no `resultWindowMaxSize` used in any place other than NetworkAddressAliasEsDAO. 

<img width="1072" alt="image" src="https://user-images.githubusercontent.com/5441976/226109506-e70dcb62-865e-4770-8217-2d30ba174d1b.png">

Reply:
This param resultWindowMaxSize is correspond the elasticsearch max limit size 10000. What I think is wrong?I am sorry for this.But this name is very confusion.
But the aim of sroll query is to query data not limit by elasticserach  limit 10000, In fact,If I didnot to change this to 20000.The other query will cause an exception. So I split them and can help the scroll query can query 20000 result,and other not scroll query will also query the data they want by the elasticsearch index max result limit.
Did I should define a othe static param?

Reply:
Are you kidding? You are contributing to an open-source project, but understand a variable from the name rather than from codes.

I am going to leave this PR, with no further steps. The conclusion is very clear, you even don't read the codes and debug the codes. Just guessing.

I am out.

Reply:
I did not just guessing, I am a coder,I think this problem is obviously to find by a test.I will do an other pull request and give the errorlog and the compare.Sorry for confusion you.